### PR TITLE
FI-430 add include search tests

### PR DESCRIPTION
**Pull request by czh-orz**
MedicationRequest and PractitionerRole both have _include search parameters that are "SHOULD" in US Core. You can test this with https://api-v8-r4.hspconsortium.org/czhour4/data and using Adams, Daniel X". 

Alternatively, use the Community tests and the open endpoint https://api-v8-r4.hspconsortium.org/czhour4/open with patient-id SMART-1288992
**Submitter:**
- [X] This pull request describes why these changes were made
- [X] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-430
- [X] Internal ticket links to this PR
- [X] Internal ticket is properly labeled (Community/Program)
- [X] Internal ticket has a justification for its Community/Program label
- [X] Code diff has been reviewed for extraneous/missing code
- [ ] Tests are included and test edge cases
- [X] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: generator/uscore/uscore_generator.rb [hunk 1]
```diff
@@ -131,6 +132,29 @@ def create_authorization_test(sequence)
...
+        search_params = first_search.nil? ? 'search_params = {}' : get_search_params(first_search[:names], sequence)
+        include_test[:test_code] = search_params
+        sequence[:include_params].each do |include|
+          included_resource = include.split(':').last.capitalize # kind of a hack, but works for now - would have to otherwise figure out resource type of target profile
+          include_test[:test_code] += %(
+                search_params['_include'] = '#{include}'
+                reply = get_resource_by_params(versioned_resource_class('#{sequence[:resource]}'), search_params)
+                assert_response_ok(reply)
+                assert_bundle_response(reply)
+                #{included_resource.downcase}_results = reply&.resource&.entry&.map(&:resource)&.select { |resource| resource.resourceType == '#{included_resource}' }
```
Comment (Jammjammjamm) [inline generator/uscore/uscore_generator.rb]:
```suggestion
                #{included_resource.downcase}_results = reply&.resource&.entry&.map(&:resource)&.any? { |resource| resource.resourceType == '#{included_resource}' }
```
Then you can just `assert #{included_resource.downcase}_results`

**Comment by czh-orz** [comment]
Good catch. This bug was introduced for the delayed sequences because of an earlier PR that made inferno look through all returned resources for a search parameter. It should be fixed in this branch now. 

Also the medicationrequest intent issue should be fixed in another PR, but you can test with PractitionerRole here. You need to run AllergyIntolerance first though.

**Comment by arscan** [comment]
Could you rebase @czh-orz

**Comment by czh-orz** [comment]
I merged with development and updated the test description to not include "SHOULD" wording.

**Pull request merged**